cdap-common/src/main/java/co/cask/cdap/common/security/AuthEnforceRewriter.java

```diff
@@ -242,8 +269,8 @@ public void visitEnd() {
             return;
           }
           AuthEnforceAnnotationNodeProcessor nodeProcessor =
-            new AuthEnforceAnnotationNodeProcessor(authEnforceAnnotationNode);
-          Map<String, Integer> paramAnnotation = processParameterAnnotationNode(parameterAnnotationNode);
+                  new AuthEnforceAnnotationNodeProcessor(authEnforceAnnotationNode);
+          Map<String, Integer> paraPositions = processParameterAnnotationNode(parameterDetails);
 
           List<EntityPartDetail> entityPartDetails = new ArrayList<>();
           for (String name : nodeProcessor.getEntities()) {
```

Comment:
can `nodeProcessor.getEntities` be empty? If so, `verifyEntityParts` will result in an exception, right (out of bounds index access)?